code/controllers/subsystems/dcs.dm

```diff
@@ -0,0 +1,54 @@
+PROCESSING_SUBSYSTEM_DEF(dcs)
+	name = "Datum Component System"
+	flags = SS_NO_INIT
+	wait = 1 SECONDS
+
+	var/list/elements_by_type = list()
+
+/datum/controller/subsystem/processing/dcs/Recover()
+	comp_lookup = SSdcs.comp_lookup
+
+/datum/controller/subsystem/processing/dcs/proc/GetElement(list/arguments)
+	var/datum/element/eletype = arguments[1]
+	var/element_id = eletype
+
+	if(!ispath(eletype, /datum/element))
+		CRASH("Attempted to instantiate [eletype] as a /datum/element")
+
+	if(initial(eletype.element_flags) & ELEMENT_BESPOKE)
+		element_id = GetIdFromArguments(arguments)
+
+	. = elements_by_type[element_id]
+	if(.)
```

Comment:
this reads like something ivan wrote at 3am while drunk

Reply:
It's actually should work exactly like that. If `elements_by_type[element_id]` exists then we return it, otherwise we set `elements_by_type[element_id]` to `new eletype` and return it